.github/workflows/build.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build Docker image
+
+on:
+  push:
+    branches: ['main']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
```

Comment:
Why is this needed? We do not use in the workflows in the monorepo https://github.com/llvm/llvm-project/blob/main/.github/workflows/build-ci-container.yml.